patches/server/0878-Added-byte-array-serialization-deserialization-for-P.patch

```diff
@@ -0,0 +1,35 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: Nex <nex@bits.team>
+Date: Thu, 24 Feb 2022 16:28:07 +0100
+Subject: [PATCH] Added byte array serialization/deserialization for
+ PersistentDataContainers
+
+
+diff --git a/src/main/java/org/bukkit/craftbukkit/persistence/CraftPersistentDataContainer.java b/src/main/java/org/bukkit/craftbukkit/persistence/CraftPersistentDataContainer.java
+index f0588bec9be09cb8273c310fb3de8bfe72dee9e5..72431c1bf8989cd248b1b18a48734916dd7bb4c7 100644
+--- a/src/main/java/org/bukkit/craftbukkit/persistence/CraftPersistentDataContainer.java
++++ b/src/main/java/org/bukkit/craftbukkit/persistence/CraftPersistentDataContainer.java
+@@ -168,5 +168,23 @@ public class CraftPersistentDataContainer implements PersistentDataContainer {
+ 
+         return this.customDataTags.containsKey(key.toString());
+     }
++
++    @Override
++    public byte[] serializeToBytes() throws java.io.IOException {
++        net.minecraft.nbt.CompoundTag root = this.toTagCompound();
++        java.io.ByteArrayOutputStream byteArrayOutput = new java.io.ByteArrayOutputStream();
++        java.io.DataOutput dataOutput = new java.io.DataOutputStream(byteArrayOutput);
++        net.minecraft.nbt.NbtIo.write(root, dataOutput);
++        return byteArrayOutput.toByteArray();
++    }
++
++    @Override
++    public void readFromBytes(byte[] bytes) throws java.io.IOException {
```

Comment:
I feel like this method should clear the pdc first, or at least provide an option to clear it before writing to it.

Reply:
Yea that's probably a good idea